client/battle/CBattleInterface.cpp

```diff
@@ -200,15 +201,9 @@ CBattleInterface::CBattleInterface(const CCreatureSet *army1, const CCreatureSet
 	}
 	else
 	{
-		auto bfieldType = curInt->cb->battleGetBattlefieldType();
-		if(!vstd::contains(graphics->battleBacks, bfieldType))
-		{
-			logGlobal->error("%s is not valid battlefield type!", static_cast<std::string>(bfieldType));
-		}
-		else
-		{
-			background = BitmapHandler::loadBitmap(graphics->battleBacks[bfieldType], false);
-		}
+		auto bfieldType = curInt->cb->battleGetBattlefieldType().getInfo();

```

Comment:
I'd suggest to check fieldType for null

Reply:
It will write an error to log about unresolved id.